### PR TITLE
PROD-3655 - Group member able to post into group indirectly even if only allowed for Moderators/Organizers. Also modify query to exclude group in activity feed

### PR DESCRIPTION
**Pull request by jitendrabanjara1991**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-3655

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-templates/bp-nouveau/includes/activity/functions.php [hunk 1]
```diff
@@ -668,3 +648,36 @@ function bp_nouveau_activity_customizer_controls( $controls = array() ) {
...
+ * Function will append join query to display group lists in the activity feed.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param string $sql From SQL statement.
+ * @param array  $r   Array of parsed arguments for the get method.
+ *
+ * @return string
+ */
+function bp_groups_get_join_sql_for_activity( $sql, $r ) {
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/functions.php:662]:
```suggestion
function bb_groups_get_join_sql_for_activity( $sql, $r ) {
```

File: src/bp-templates/bp-nouveau/includes/activity/functions.php [hunk 2]
```diff
@@ -668,3 +648,36 @@ function bp_nouveau_activity_customizer_controls( $controls = array() ) {
...
+ * Function will append where condition to display group lists in the activity feed.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param array $where_conditions Where conditions SQL statement.
+ * @param array $r                Array of parsed arguments for the get method.
+ *
+ * @return mixed
+ */
+function bp_groups_get_where_conditions_for_activity( $where_conditions, $r ) {
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/functions.php:679]:
```suggestion
function bb_groups_get_where_conditions_for_activity( $where_conditions, $r ) {
```

File: src/bp-templates/bp-nouveau/includes/activity/functions.php [hunk 3]
```diff
@@ -201,31 +201,11 @@ function bp_nouveau_activity_localize_scripts( $params = array() ) {
...
-							$exclude_groups[] = $exclude_group_id;
-						}
-					}
-				}
-				
-				if ( ! empty( $exclude_groups ) ){
-					$group_args['exclude'] = $exclude_groups;
-				}
+				add_filter( 'bp_groups_get_join_sql', 'bp_groups_get_join_sql_for_activity', 10, 2 );
+				add_filter( 'bp_groups_get_where_conditions', 'bp_groups_get_where_conditions_for_activity', 10, 2 );
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/functions.php:205]:
```suggestion
				add_filter( 'bp_groups_get_join_sql', 'bb_groups_get_join_sql_for_activity', 10, 2 );
				add_filter( 'bp_groups_get_where_conditions', 'bb_groups_get_where_conditions_for_activity', 10, 2 );
```

File: src/bp-templates/bp-nouveau/includes/activity/functions.php [hunk 4]
```diff
@@ -201,31 +201,11 @@ function bp_nouveau_activity_localize_scripts( $params = array() ) {
...
 				$group_query_cache[ $cache_key ] = groups_get_groups( $group_args );
+				remove_filter( 'bp_groups_get_join_sql', 'bp_groups_get_join_sql_for_activity', 10, 2 );
+				remove_filter( 'bp_groups_get_where_conditions', 'bp_groups_get_where_conditions_for_activity', 10, 2 );
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/activity/functions.php:208]:
```suggestion
				remove_filter( 'bp_groups_get_join_sql', 'bb_groups_get_join_sql_for_activity', 10, 2 );
				remove_filter( 'bb_groups_get_where_conditions', 'bp_groups_get_where_conditions_for_activity', 10, 2 );
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need changes

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**